controller/canary_test.go

```diff
@@ -379,9 +379,9 @@ func TestCanaryRolloutCreateFirstReplicasetNoSteps(t *testing.T) {
 	expectedPatch := `{
 		"status":{
 			"canary":{
-				"stableRS":"895c6c4f9"
+				"stableRS":"` + rs.Labels[v1alpha1.DefaultRolloutUniqueLabelKey] + `"
 			},
-			"currentPodHash":"895c6c4f9",
+			"currentPodHash":"` + rs.Labels[v1alpha1.DefaultRolloutUniqueLabelKey] + `",
 			"conditions": %s
```

Comment:
I feel like we should be consistent with how we generate our expectedPatches and not have two ways of injecting variables. I like your approach more than using Sprintf to inject variables though and I think we should open another PR after this one to converge to a standard.

Reply:
Will be covered as part of: https://github.com/argoproj/argo-rollouts/issues/107